### PR TITLE
[ATOM-716] Port capability to select which custom actions should receive domain to 3.4.x

### PR DESCRIPTION
**Pull request by radovanZRasa**
**Proposed changes**:
Port support to select which custom actions should receive domain.
Original implementation: [PR-12013](https://github.com/RasaHQ/rasa/pull/12013)

**Status (please check what you already did)**:
- [x] added some tests for the functionality
- [x] updated the documentation
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [ ] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

**Review by sanchariGr — APPROVED** [review]

File: docs/docs/domain.mdx [hunk 1]
```diff
@@ -794,6 +794,31 @@ For example, an action could:
...
+:::info New in 3.4.3
+You can control if an action should receive a domain or not.
+:::
+
+To do this you must first enable selective domain in you endpoint configuration for
+`action_endpoint` in `endpoints.yml`.
+```YAML
+# endpoints.yml
+action_endpoint:
+ url: "http://localhost:5055/webhook" #URL to your action server1
```
Comment (ancalita) [inline docs/docs/domain.mdx:807]: Added whitespace + removed 1
```suggestion
 url: "http://localhost:5055/webhook" # URL to your action server
```

File: rasa/shared/core/domain.py [hunk 2]
```diff
@@ -1858,6 +1881,35 @@ def __repr__(self) -> Text:
...
+    @staticmethod
+    def _collect_actions_which_explicitly_need_domain(
+        actions: List[Union[Text, Dict[Text, Any]]]
+    ) -> List[Text]:
+        action_names: List[Text] = []
+
+        for action in actions:
+            if isinstance(action, dict):
+                for action_name, action_config in action.items():
+                    should_send_domain = action_config.get("send_domain", False)
```
Comment (ancalita) [inline rasa/shared/core/domain.py:1907]: Optional: extract these constant strings to the constants module.

File: tests/shared/utils/test_validation.py [hunk 3]
```diff
@@ -33,6 +33,96 @@ def test_validate_yaml_schema(file, schema):
...
+
+        actions:
+          - utter_default: {send_domain: ""}
+        """,
+            DOMAIN_SCHEMA_FILE,
+        ),
+    ],
+)
+def test_invalid_send_domain_value_in_actions(content: Text, schema: Text):
+    # should raise no exception
```
Comment (ancalita) [inline tests/shared/utils/test_validation.py:121]: I'm confused by this comment, should it be removed considering this test should raise an exception.

**Review by ancalita — APPROVED** [review]
Looks good 👍🏻 Picked up some minor clean-up items.

**Pull request merged**